### PR TITLE
Email list jsonfield

### PR DESCRIPTION
**Pull request by nickpell**
Previously, billing contact emails where represented as strings concatenated together with ",".

This change means that the field can be accessed directly as a list, which helps clean up usages of the field.  Once we get to Django 1.9, I'd then like to change the json_field to an ArrayField for a more specific schema and Django email validation, but this change gets most of the benefits when accessing the email list in the source code.

@benrudolph @proteusvacuum cc: @emord

### CODE REVIEW

File: corehq/apps/accounting/forms.py [hunk 1]
```diff
@@ -121,7 +119,7 @@ def __init__(self, account, *args, **kwargs):
                 'name': account.name,
                 'salesforce_account_id': account.salesforce_account_id,
                 'currency': account.currency.code,
-                'emails': contact_info.emails,
+                'email_list': ','.join(contact_info.email_list),
```
Comment (benrudolph) [inline corehq/apps/accounting/forms.py]: `email_list` could be a bit confusing considering it's a string not a list, but admittedly don't really have a better name for it

Comment (benrudolph) [inline corehq/apps/accounting/forms.py]: ah nvm it is a list!

File: corehq/apps/accounting/migrations/0015_datamigration_email_list.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+def copy_emails_to_email_list(apps, schema_editor):
+    BillingContactInfo = apps.get_model('accounting', 'BillingContactInfo')
+
+    for contact_info in BillingContactInfo.objects.all():
+        if contact_info.emails:
+            contact_info.email_list = contact_info.emails.split(',')
+        else:
+            contact_info.email_list = []
+        contact_info.save()
```
Comment (benrudolph) [inline corehq/apps/accounting/migrations/0015_datamigration_email_list.py]: assuming there aren't too many of these so it's probably fine, but this could really do a number on the db if there are a significant number of records

File: corehq/apps/accounting/models.py [hunk 3]
```diff
@@ -536,6 +538,11 @@ class BillingContactInfo(models.Model):
     class Meta:
         app_label = 'accounting'
 
+    def __init__(self, *args, **kwargs):
+        super(BillingContactInfo, self).__init__(*args, **kwargs)
+        if self.email_list == '[]':
```
Comment (benrudolph) [inline corehq/apps/accounting/models.py]: why is this necessary?

**Comment by benrudolph** [comment]
nice i'm liking this. test failure looks real:
```
======================================================================
ERROR: test_manual_adjustment (corehq.apps.accounting.tests.test_forms.TestAdjustBalanceForm)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/accounting/tests/test_forms.py", line 16, in setUp
    super(TestAdjustBalanceForm, self).setUp()
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/accounting/tests/test_invoicing.py", line 32, in setUp
    self.dimagi_user, self.billing_contact)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/accounting/generator.py", line 81, in billing_account
    billing_contact = arbitrary_contact_info(billing_account, web_user_contact)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/accounting/generator.py", line 98, in arbitrary_contact_info
    country="US",
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/accounting/models.py", line 542, in __init__
    super(BillingContactInfo, self).__init__(*args, **kwargs)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/db/models/base.py", line 452, in __init__
    raise TypeError("'%s' is an invalid keyword argument for this function" % list(kwargs)[0])
TypeError: 'emails' is an invalid keyword argument for this function
```

File: corehq/apps/accounting/models.py (hunk 3 above)
Comment (nickpell) [inline corehq/apps/accounting/models.py]: ```json_field``` has a weird condition where it defaults to ```'[]'``` instead of ```[]```. So this corrects that.

Comment (benrudolph) [inline corehq/apps/accounting/models.py]: :+1:

File: corehq/apps/accounting/migrations/0015_datamigration_email_list.py (hunk 2 above)
Comment (nickpell) [inline corehq/apps/accounting/migrations/0015_datamigration_email_list.py]: so far I've been ok writing billing migrations with a similar workload.  but yeah, agreed

**Comment by nickpell** [comment]
:white_check_mark:

File: corehq/apps/accounting/models.py (hunk 3 above)
Comment (snopoke) [inline corehq/apps/accounting/models.py]: have you tried setting the default to `list` instead of `[]`

Comment (nickpell) [inline corehq/apps/accounting/models.py]: odd, now I'm unable to reproduce.  Will investigate further.
